PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = 'ir.actions.report'
-    _inherit = 'product.document'
-    _inherit = 'sale.order'
-    _inherit = 'sale.order.line'
-    _inherit = 'sale.order.template'

Total Changes: 12

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/sale_pdf_quote_builder/models/ir_actions_report.py
+++ b/addons/sale_pdf_quote_builder/models/ir_actions_report.py
@@ -10,7 +10,7 @@ from odoo.tools.pdf import NameObject, NumberObject, PdfFileReader, PdfFileWrite
 
 
 class IrActionsReport(models.Model):
-    _inherit = 'ir.actions.report'
+    _inherit = ['ir.actions.report']
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         """Override to add and fill headers, footers and product documents to the sale quotation."""

--- a/addons/sale_pdf_quote_builder/models/product_document.py
+++ b/addons/sale_pdf_quote_builder/models/product_document.py
@@ -9,7 +9,7 @@ from odoo.addons.sale_pdf_quote_builder import utils
 
 
 class ProductDocument(models.Model):
-    _inherit = 'product.document'
+    _inherit = ['product.document']
 
     attached_on_sale = fields.Selection(
         selection_add=[('inside', "Inside quote pdf")],

--- a/addons/sale_pdf_quote_builder/models/quotation_document.py
+++ b/addons/sale_pdf_quote_builder/models/quotation_document.py
@@ -9,7 +9,6 @@ from odoo.addons.sale_pdf_quote_builder import utils
 
 
 class QuotationDocument(models.Model):
-    _name = 'quotation.document'
     _description = "Quotation's Headers & Footers"
     _inherits = {
         'ir.attachment': 'ir_attachment_id',

--- a/addons/sale_pdf_quote_builder/models/sale_order.py
+++ b/addons/sale_pdf_quote_builder/models/sale_order.py
@@ -6,7 +6,7 @@ from odoo import _, api, fields, models
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = ['sale.order']
 
     available_product_document_ids = fields.Many2many(
         string="Available Product Documents",

--- a/addons/sale_pdf_quote_builder/models/sale_order_line.py
+++ b/addons/sale_pdf_quote_builder/models/sale_order_line.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = 'sale.order.line'
+    _inherit = ['sale.order.line']
 
     available_product_document_ids = fields.Many2many(
         string="Available Product Documents",

--- a/addons/sale_pdf_quote_builder/models/sale_order_template.py
+++ b/addons/sale_pdf_quote_builder/models/sale_order_template.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class SaleOrderTemplate(models.Model):
-    _inherit = 'sale.order.template'
+    _inherit = ['sale.order.template']
 
     quotation_document_ids = fields.Many2many(
         string="Headers and footers",

--- a/addons/sale_pdf_quote_builder/models/sale_pdf_form_field.py
+++ b/addons/sale_pdf_quote_builder/models/sale_pdf_form_field.py
@@ -9,7 +9,6 @@ from odoo.addons.sale_pdf_quote_builder import utils
 
 
 class SalePdfFormField(models.Model):
-    _name = 'sale.pdf.form.field'
     _description = "Form fields of inside quotation documents."
     _order = 'name'
 
